src/strategies/schedule.py

```diff
@@ -18,13 +28,343 @@ def __init__(self, constants, start_time, **kwargs):
+                assert cs.max_power >= cs.current_power - self.EPS, (
+                    "{} - {} over maximum load ({} > {})".format(
+                        self.current_time, cs_id, cs.current_power, cs.max_power))
+
+                # pass on unused allocated power to next vehicle
+                extra_power = max(power_allocated_for_vehicle - avg_power, 0)
+                # once every vehicle had a chance to charge and there is no
+                # extra power to be distributed, stop charging process
+                if i >= n_vehicles and extra_power < self.EPS:
+                    break
+
+                # vehicle didnt get to charge, going to the back of the line
+                # allocated power + extra power might be enough to charge in
+                # a second try
+                if (cs.max_power - cs.current_power > self.EPS and
+                        remaining_power_on_schedule >= cs.min_power and
+                        remaining_power_on_schedule >= vehicle.vehicle_type.min_charging_power and
+                        vehicle.get_delta_soc() > self.EPS):
+                    vehicles.append((vehicle_id, energy_needed))
+
+        # last timestep of core standing time
+        if dt_to_end_core_standing_time <= self.interval:
+            # In case not all vehicles are satisfied, allow
+            # charging over schedule outside of core standing time
+            if not all([v.desired_soc - v.battery.soc < self.EPS
+                        for v in self.world_state.vehicles.values()]):
+                self.overcharge_necessary = True
+
+            self.currently_in_core_standing_time = False
+
+        return charging_stations
+
+    def charge_cars_after_core_standing_time(self, charging_stations):
+        gc = list(self.world_state.grid_connectors.values())[0]  # only 1 GC supported
+
+        total_power = gc.cur_max_power - gc.get_current_load()
+        vehicles = self.world_state.vehicles.values()
+
+        power_needed = []
+        for vehicle in vehicles:
+            if vehicle.connected_charging_station is None:
+                continue
+            soc_needed = vehicle.desired_soc - vehicle.battery.soc
+            power_needed.append(soc_needed * vehicle.battery.capacity)
+
+        if sum(power_needed) < self.EPS:
+            # cars fully charged
+            self.overcharge_necessary = False
+            return charging_stations
+
+        if total_power < self.EPS:
+            # grid connector maxed out
+            return charging_stations
+
+        # charge vehicles balanced until departure
+        for vehicle in vehicles:
+            cs_id = vehicle.connected_charging_station
+            if cs_id is None:
+                continue
+            cs = self.world_state.charging_stations[cs_id]
+            time_until_departure = vehicle.estimated_time_of_departure - self.current_time
+            power = self.sim_charging_process(
```

Comment:
instead of the generic sim_charging, make clear that the cars are charged balanced (`sim_balanced_charging`, for example).